docker/Dockerfile

```diff
@@ -1,7 +1,17 @@
-FROM alpine:latest
+FROM golang:1.16-alpine AS builder
 
-COPY restic /usr/bin
+WORKDIR /go/src/github.com/restic/restic
+
+COPY go.mod go.sum ./
```

Comment:
I  guess the idea here is to allow caching the dependencies? Please add a short comment

Reply:
Yes. As long as `go.mod` and `go.sum` don't change, the result of these steps is taken from the cache rather than executing the commands again.

Reply:
I'm not a friend of comments, but I've added one on your request.